tools/stale_pr_date_setter.py

```diff
@@ -0,0 +1,272 @@
+"""
+Tarantool Lango team PR last activity date setter.
+
+This script makes use of GitHub GraphQL API to check the activity
+on pull requests listed on the specified board and update the
+last activity date correspondingly.
+
+Parameters are passed via environment variables.
+Available parameters:
+    - LANGO_REVIEW_BOARD_TOKEN: GitHub token with read/write rights
+    for the specified project and read rights for PRs in that project.
+    - ORGANIZATION: Name of the organization that project belongs to.
+    - PROJECT_ID: ID of the GitHub project.
+
+NB: This script is suitable only for GitHub ProjectV2.
+"""
+import json
+import logging
+import os
+import sys
+from datetime import datetime, timezone
+from enum import Enum
+
+import requests
+
+
+class Exitcode(Enum):
+    ERR_PROJECT = 1
+    ERR_FIELD = 2
+    ERR_PR = 3
+    ERR_PARAMS = 4
+
+
+logging.basicConfig()
+logger = logging.getLogger('review_date_setter')
+logger.setLevel(logging.INFO)
+
+try:
+    TOKEN = os.environ['LANGO_REVIEW_BOARD_TOKEN']
+    ORGANIZATION = os.environ['ORGANIZATION']
+    PROJECT_ID = int(os.environ['PROJECT_ID'])
+except KeyError:
+    logger.critical('No parameters provided')
+    sys.exit(Exitcode.ERR_PARAMS)
+
+HEADERS = {'Authorization': 'Bearer {token}'.format(token=TOKEN)}
+ITEM_DATE_FIELD_NAME = 'No response since'
+URL = 'https://api.github.com/graphql'
+
+PROJECT_ID_REQUEST = """
+query
+{
+    organization(login: "%s")
+    {
+        projectV2(number: %d)
+        {
+            id
+        }
+    }
+}
+"""
+
+DATEFIELD_ID_REQUEST = """
+query
+{
+    node(id: "%s")
+    {
+        ... on ProjectV2
+        {
+            field(name: "%s")
+            {
+                ... on ProjectV2FieldCommon
+                {
+                    id
+                    name
+                }
+            }
+        }
+    }
+}
+"""
+
+ITEMS_REQUEST = """
+query
+{
+    node(id: "%s")
+    {
+        ... on ProjectV2
+        {
+            items(first: 20 after: "%s")
+            {
+                pageInfo
+                {
+                    startCursor
+                    hasNextPage
+                    endCursor
+                }
+
+                nodes
+                {
+                    id
+                    fieldValueByName(name: "%s")
+                    {
+                        ... on ProjectV2ItemFieldDateValue
+                        {
+                            date
+                        }
+                    }
+                    content
+                    {
+                        ...on PullRequest
```

Comment:
Checkpatch complains:

```
ERROR:TYPO_SPELLING: 'PullRequest' may be misspelled - perhaps 'Pull request'?
#166: FILE: tools/stale_pr_date_setter.py:111:
+                        ...on PullRequest
                              ^^^^^^^^^^^
```

Don't know what is better to do here, just noted.

Reply:
I think it can be ignored.

Reply:
Fixed.